### DISCUSSION TITLE
Final decision: How do you want to access preview VSIX releases?

### DISCUSSION THREAD

**Discussion by andrewleader** [microsoft/WindowsAppSDK#742] [Final decisions]
**Final decision:** We decided we'll publish a side-by-side "Preview" VSIX on the official VS Marketplace. This means you can have both the GA VSIX installed and the "Preview" VSIX installed side-by-side (and all project/item templates within the Preview VSIX will be labeled so you can differentiate between the two).

We're discussing how we should release previews of our VSIX's. Occasionally we'll have some new project templates or item templates in the VSIX that aren't yet GA (and potentially won't be GA for several releases). **How would you like to access these preview VSIX releases?**

We could...

* Make the "Project Reunion (Preview)" VSIX available directly **on the [Visual Studio Marketplace](https://marketplace.visualstudio.com/vs)**. Thus there's two listings on the Marketplace, "Project Reunion", and "Project Reunion (Preview)".
* Or make the "Project Reunion (Preview)" VSIX available **elsewhere, like as a [GitHub Release asset](https://github.com/microsoft/ProjectReunion/releases)**. Thus, there's only one listing on the Marketplace, and previews will be accessed via a direct download.

What would you prefer and why?

**Comment by riverar** [comment]
Considering we are moving towards [at least 3 different release channels](https://github.com/microsoft/ProjectReunion/discussions/637), Visual Studio Marketplace feels like the only option that makes sense. GitHub Releases UX isn't great, only _really_ allows you to surface _pre-release_ and _release_ assets, uses pre-release naming that cannot be customized, branch tagging will get wonky fast, and doesn't integrate with Visual Studio.

**Comment by mrlacey** [comment]
Please consider using the https://www.vsixgallery.com/ for preview releases as many other extensions already do.

It will avoid any potential confusion by having multiple entries in the marketplace and still provides an easy way to get updates displayed and notified inside Visual Studio, while GitHub assets wouldn't.

**Reply by andrewleader** [reply]
Thanks Matt! Hmm, do you have a specific example of a preview release on that VSIX Gallery? I searched the word "preview" and didn't see anything appear there, but if this is the most common way of releasing previews for extensions, would love to learn that! Seems great that it can appear inside VS too.

**Reply by andrewleader** [reply]
Ah I just saw your internal email, I'll reach out to Mads 😊

**Reply by riverar** [reply]
How is this better than the official marketplace?

**Reply by mrlacey** [reply]
> How is this better than the official marketplace?

It's not. It's different but offers most of the same functionality. The benefit here is having only the official/stable release in the marketplace so that anyone looking there doesn't accidentally get a preview release. 
Having preview releases in a separate place enables them to be found by the people who want them.

**Reply by mrlacey** [reply]
>  I searched the word "preview" and didn't see anything appear there, but if this is the most common way of releasing previews for extensions, would love to learn that! 

The convention with use of the open VSIX gallery is that extension authors post CI builds there. It's not common for them to change the name. A different version number is typically used to identify a preview build.

**Comment by andrewleader** [comment]
**Update:** We decided we'll publish a side-by-side "Preview" VSIX on the official VS Marketplace. This means you can have both the GA VSIX installed and the "Preview" VSIX installed side-by-side (and all project/item templates within the Preview VSIX will be labeled so you can differentiate between the two).

Matt - Thank you for sharing insights on the Open VSIX Gallery, that was super insightful. In our case though, these "previews" really aren't previews of our next upcoming release, they're more a "long-lead/experimental" release that includes templates that potentially won't GA for multiple release cycles (so that's why having just a different version number doesn't exactly work, this is technically a separate "release channel", not really just a preview of the upcoming next release (I know, the term "Preview" is therefore confusing, but we're getting close on closing on our final release channels and hopefully can have better terminology and clarity around this soon).

We're hopeful someday soon we won't need side-by-side previews as our VSIX has fewer and fewer features that aren't "long-lead/experimental", so it's possible that in the future we'll delete this preview VSIX from the marketplace if/when it no longer becomes relevant/necessary (as many of you have told us, "why should I need a new VSIX to use the new features" - fully agreed).
